microsoft/PowerToys
# [FancyZones] Use Direct2D to draw

## Summary of the Pull Request

* Read the title
* Removed flashZones and references to it

## PR Checklist
* [x] Applies to #4073

## Info on Pull Request

Refactored existing code for drawing and used Direct2D to do drawing. Ensured we have fine control over what and when something is drawn by using a dedicated thread and securing data access using mutexes and condition variables.

## Validation Steps Performed

Create a zone layout with many rows/columns and shift-move a window quickly across the screen. You should see absolutely no stuttering of any kind.
